packages/atlas/src/views/viewer/VideoView/CommentsSection.tsx

```diff
@@ -434,23 +178,37 @@ export const CommentsSection: React.FC<CommentsSectionProps> = ({ disabled, vide
       <CommentInput
         memberAvatarUrl={memberAvatarUrl}
         isMemberAvatarLoading={authorized ? isMemberAvatarLoading : false}
-        processing={commentInputIsProcessingCollection.has(COMMENT_BOX_ID)}
+        processing={commentInputIsProcessing}
         readOnly={!activeMemberId}
         memberHandle={activeMembership?.handle}
-        onFocus={() => !activeMemberId && openSignInDialog({ onConfirm: signIn })}
-        onComment={() => handleComment({ commentInputId: COMMENT_BOX_ID })}
-        value={commentInputTextCollection.get(COMMENT_BOX_ID) ?? ''}
-        hasInitialValueChanged={!!commentInputTextCollection.get(COMMENT_BOX_ID)}
+        value={commentInputText}
+        hasInitialValueChanged={!!commentInputText}
         withoutOutlineBox
-        onChange={(e) => setCommentInputText({ commentId: COMMENT_BOX_ID, comment: e.target.value })}
+        onFocus={() => !activeMemberId && openSignInDialog({ onConfirm: signIn })}
+        onComment={() => handleComment()}
+        onChange={(e) => setCommentInputText(e.target.value)}
       />
       {comments && !comments.length && (
         <EmptyFallback title="Be the first to comment" subtitle="Nobody has left a comment under this video yet." />
       )}
       <CommentWrapper>
         {commentsLoading
-          ? placeholderItems.map((_, idx) => <Comment key={idx} type="default" loading />)
-          : memoizedComments}
+          ? placeholderItems.map((_, idx) => <Comment key={idx} />)
+          : filteredComments?.map((comment, idx) => (
+              <CommentThread
+                key={`${comment.id}-${idx}`}
+                commentId={comment.id}
+                memberUrl={absoluteRoutes.viewer.member(comment.author.handle)}
+                video={video}
+                replies={comment.replies}
+                repliesCount={comment.repliesCount}
+                highlightedCommentId={highlightedCommentId}
+                indented={!!comment.parentCommentId && comment.id === commentIdQueryParam}
+                setOriginalComment={setOriginalComment}
+                setHighlightedCommentId={setHighlightedCommentId}
+                setShowEditHistory={setShowEditHistory}
+              />
+            ))}
       </CommentWrapper>
       <DialogModal
         size="medium"
```

Comment:
Couldn't this be a part of `Comment` as well? Then we can avoid passing some state around

Reply:
actually yes